plugins/defaultplugins/aclplugin/vppdump/dump_vppcalls.go

```diff
@@ -211,6 +469,15 @@ func getIPRule(r acl_api.ACLRule) *acl.AccessLists_Acl_Rule_Matches_IpRule {
 	return &ipRule
 }
 
+func getMACIPRuleMatches(rule acl_api.MacipACLRule) *acl.AccessLists_Acl_Rule_Matches_MacIpRule {
+	return &acl.AccessLists_Acl_Rule_Matches_MacIpRule{
+		SourceAddress:        net.IP(rule.SrcIPAddr[:4]).To4().String(),
```

Comment:
This will panic if SrcIPAddr is zero length. Why not just use whole array there? If it contains IPv6 it would output wrong thing anyway, since you cannot just take first 4 bytes from IPv6 and make it IPv4 anyway.

Reply:
Yes, it can, but the length of the SrcIPAddr should never be shorter than 16 bytes (field is defined by vpp api). 
Why not just use the whole array here? Because vpp will always fill the field with 16 bytes - it means that the IPv4 address is written as [192 168 0 1 0 0 0 0 0 0 0 0 0 0 0 0] (for example).

Now, let's parse 192.168.0.1 []byte to string: 
net.IP(rule.SrcIPAddr[:4]).To4().String() = "192.168.0.1"
net.IP(rule.SrcIPAddr[:4]).String() = "192.168.0.1"
net.IP(rule.SrcIPAddr).To4().String() = nil
net.IP(rule.SrcIPAddr).String() = "c0a8:1::"

It means I need to take just first four bytes if IPv4 address is used.

However IPv6 resolution was incorrect here, I'll fix it

Reply:
But you are "taking first four bytes" by using `To4()`.